spark2.3/build-assembly.sbt

```diff
@@ -19,6 +19,7 @@ assemblyMergeStrategy in assembly := {
   case PathList("META-INF", "io.netty.versions.properties")  => MergeStrategy.first
   // db-engine-2.8.0.1926.jar VS splice_encoding-2.8.0.1926.jar
   case PathList("com", "splicemachine", "utils", _*)  => MergeStrategy.first
+  case PathList("net", "jcip", "annotations", "GuardedBy.class") => MergeStrategy.last
```

Comment:
Why is this needed to "fix imports for build"?

Reply:
This was done just to sync the spark2.3 code with [the same change previously done in the spark2.4 code](https://github.com/splicemachine/splice-machine-spark-connector/commit/b01c0b86400ffb7fbdffcae56dce68778e37f774).